lib/Common/Memory/Allocator.h

```diff
@@ -382,16 +384,19 @@ void DestructArray(size_t count, T* obj)
 template <typename TAllocator, typename T>
 void DeleteArray(typename AllocatorInfo<TAllocator, T>::AllocatorType * allocator, size_t count, T * obj)
 {
-    if (count == 0)
+    if (count == 0 && TAllocator::FakeZeroLengthArray)
```

Comment:
Should this be AllocatorInfo<TAllocator, T>::AllocatorType::FakeZeroLengthArray instead?